Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -188,7 +188,36 @@ extension BrowserViewController: WKNavigationDelegate {
         let isPrivateBrowsing = PrivateBrowsingManager.shared.isPrivateBrowsing
         
         // Check if custom user scripts must be added to the web view.
-        tabManager[webView]?.userScriptManager?.handleDomainUserScript(for: url)
+        let tab = tabManager[webView]
+        tab?.userScriptManager?.handleDomainUserScript(for: url)
+        
+        // Brave Search logic.
```

Comment:
is this only called on top-level navigations? (i assume so, just double checking)

Reply:
added top-level check